mongo/collection.go

```diff
@@ -326,15 +326,13 @@ func (coll *Collection) insert(ctx context.Context, documents []interface{},
 func (coll *Collection) InsertOne(ctx context.Context, document interface{},
 	opts ...*options.InsertOneOptions) (*InsertOneResult, error) {
 
-	imOpts := make([]*options.InsertManyOptions, len(opts))
-	for i, opt := range opts {
-		imo := options.InsertMany()
-		if opt != nil && opt.BypassDocumentValidation != nil && *opt.BypassDocumentValidation {
-			imo = imo.SetBypassDocumentValidation(*opt.BypassDocumentValidation)
-		}
-		imOpts[i] = imo
+	ioOpts := options.MergeInsertOneOptions(opts...)
+	var imOpts options.InsertManyOptions
```

Comment:
This can be declared as `imOpts := &options.InsertMany()`. For the case of `InsertMany`, this is probably a no-op, but in other cases, the options constructors can be responsible for setting defaults, so this is the idiomatic way to declare options.

Reply:
Good point, done.